### PR TITLE
[Search] Add 2024-05-01-preview features

### LINKED ISSUE #43005: Add Search embedding sample for reduced embedding dimensions
Add a sample for [reducing embedding dimensions](https://platform.openai.com/docs/guides/embeddings/use-cases). Provide cases for scenarios where embeddings are being indexed client-side and with the vectorizer skill.

Additionally, give some overview on the reduced dimensions when the embedding is or is not normalized. This is a case where the vector search algorithm metric may change when searching (dot product when normalized and cosine when not normalized).

### LINKED ISSUE #43552: [Flaky Test] `Champion_OneShotUpload` and `CanContinueWithNextPage`
### Library name and version

Azure.Search.Documents

### Describe the bug

Log: https://dev.azure.com/azure-sdk/public/_build/results?buildId=3718646&view=logs&j=34b93243-28ae-5187-116f-8161d4cfbe05&t=e08c007d-4f8f-562f-893c-283c0d3e03ee
Log: https://dev.azure.com/azure-sdk/public/_build/results?buildId=3718705&view=logs&jobId=5c3f95c0-a505-5338-4c9d-91c19e24e131&j=5c3f95c0-a505-5338-4c9d-91c19e24e131&t=13d1ca74-5235-5ba0-70f5-5accaff71344

this case is always timing out which blocks PR to merge.

### Expected behavior

Test should pass.

### Actual behavior

This test will time out.

### Reproduction Steps

Remove the `Ignore` attribute and see whether it could pass on our pipeline

### Environment

_No response_

### PR DESCRIPTION
**Pull request by ShivangiReja**
This PR adds the following preview features for `2024-05-01-preview` API version. 

- Added support for new embedding models `text-embedding-3-small` and `text-embedding-3-large`, as part of our existing `AzureOpenAIVectorizer` and `AzureOpenAIEmbeddingSkill` features. 
- Added support for `AIServicesVisionVectorizer` and `VisionVectorizeSkill` to generate embeddings for an image or text using the Azure AI Services Vision Vectorize API.
- Added support for `AzureMachineLearningVectorizer`, allowing users to specify an Azure Machine Learning endpoint deployed via the Azure AI Studio Model Catalog to generate vector embeddings of query strings.
- Added support for `byte` to `SearchFieldDataType`.

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/43552
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/43005

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch_UsingReducedEmbeddings.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+In order to get the reduced embeddings using either the `text-embedding-3-small` or `text-embedding-3-large` models, it is necessary to include the `Dimensions` parameter. This parameter enables customers to define the desired number of dimensions for the output vector. Therefore, for `AzureOpenAIVectorizer`, we will retrieve the `VectorSearchDimensions` that is already specified in the corresponding index field definition. However, to ensure that dimensions are only passed along in the vectorizer for a model that supports it, we need to pass a required property named `ModelName`. This property enables the service to determine which model we are using, and dimensions will only be passed along when it is for a known supported model name.
```
Comment (alzimmermsft) [inline sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch_UsingReducedEmbeddings.md:9]:
```suggestion
In order to get the reduced embeddings using either the `text-embedding-3-small` or `text-embedding-3-large` models, it is necessary to include the `Dimensions` parameter. This parameter configures the desired number of dimensions for the output vector. Therefore, for `AzureOpenAIVectorizer`, we will retrieve the `VectorSearchDimensions` that is already specified in the corresponding index field definition. However, to ensure that dimensions are only passed along in the vectorizer for a model that supports it, we need to pass a required property named `ModelName`. This property enables the service to determine which model we are using, and dimensions will only be passed along when it is for a known supported model name.
```

Usage of customer in that sentence felt off as we didn't reference customer anywhere else

File: sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch_UsingReducedEmbeddings.md
Comment (alzimmermsft) [inline sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch_UsingReducedEmbeddings.md:1]: Should this be Sample08?

File: sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch_UsingSemanticHybridQuery.md [hunk 2]
```diff
@@ -96,7 +96,7 @@ public static ReadOnlyMemory<float> GetEmbeddings(string input)
     AzureKeyCredential credential = new AzureKeyCredential(key);
 
     OpenAIClient openAIClient = new OpenAIClient(endpoint, credential);
-    EmbeddingsOptions embeddingsOptions = new("EmbeddingsModelName", new string[] { input });
+    EmbeddingsOptions embeddingsOptions = new("text-embedding-ada-002", new string[] { input });
```
Comment (alzimmermsft) [inline sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch_UsingSemanticHybridQuery.md:99]: Should we use the new enum added that contains this value?

File: sdk/search/Azure.Search.Documents/src/Generated/Models/SearchServiceLimits.cs [hunk 3]
```diff
@@ -20,12 +20,14 @@ internal SearchServiceLimits()
...
         /// <param name="maxComplexObjectsInCollectionsPerDocument"> The maximum number of objects in complex collections allowed per document. </param>
-        internal SearchServiceLimits(int? maxFieldsPerIndex, int? maxFieldNestingDepthPerIndex, int? maxComplexCollectionFieldsPerIndex, int? maxComplexObjectsInCollectionsPerDocument)
+        /// <param name="maxStoragePerIndex"> The maximum amount of storage in bytes allowed per index. </param>
+        internal SearchServiceLimits(int? maxFieldsPerIndex, int? maxFieldNestingDepthPerIndex, int? maxComplexCollectionFieldsPerIndex, int? maxComplexObjectsInCollectionsPerDocument, long? maxStoragePerIndex)
         {
             MaxFieldsPerIndex = maxFieldsPerIndex;
             MaxFieldNestingDepthPerIndex = maxFieldNestingDepthPerIndex;
             MaxComplexCollectionFieldsPerIndex = maxComplexCollectionFieldsPerIndex;
             MaxComplexObjectsInCollectionsPerDocument = maxComplexObjectsInCollectionsPerDocument;
+            MaxStoragePerIndex = maxStoragePerIndex;
```
Comment (alzimmermsft) [inline sdk/search/Azure.Search.Documents/src/Generated/Models/SearchServiceLimits.cs:30]: nit: Swagger transform rename to `MaxStoragerPerIndexInBytes`

File: sdk/search/Azure.Search.Documents/tests/FieldBuilderTests.cs [hunk 4]
```diff
@@ -611,7 +618,7 @@ private class ModelWithUnsupportedEnumerableType
             public string ID { get; set; }
 
             [SimpleField(IsFilterable = true)]
-            public IEnumerable<byte> Buffer { get; set; }
+            public IEnumerable<char> Buffer { get; set; }
```
Comment (alzimmermsft) [inline sdk/search/Azure.Search.Documents/tests/FieldBuilderTests.cs:621]: Any reason this changed from byte to char?

Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/tests/FieldBuilderTests.cs:621]: Yes, this model is for unsupported EnumerableType. Now that we support byte, I updated the type which is unsupported.

File: sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch_UsingReducedEmbeddings.md
Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch_UsingReducedEmbeddings.md:1]: No, all the vector search samples are sub-samples of Sample07_VectorSearch.md listed like this: [link](https://github.com/Azure/azure-sdk-for-net/blob/5a479328e90c2b81941231a9b6e39b08094e2765/sdk/search/Azure.Search.Documents/samples/Sample07_VectorSearch.md).

**Comment by pallavit** [comment]
LGTM. /cc: @christothes could you please take a look at this as well.

**Review by pallavit — APPROVED** [review]

**Review by christothes — APPROVED** [review]

**Pull request merged**